os/drivers/serial/serial.c

```diff
@@ -266,6 +360,9 @@ static int uart_putxmitchar(FAR uart_dev_t *dev, int ch, bool oktoblock)
 		if (nexthead != dev->xmit.tail) {
 			dev->xmit.buffer[dev->xmit.head] = ch;
 			dev->xmit.head = nexthead;
+#ifdef CONFIG_DYNAMIC_LOGS_STORING
+			add_char_to_log_net(ch);
+#endif
```

Comment:
Is this cover all of logs? As far as I know, when hw tx buffer is full, `uart_xmitchars` function copies ch from sw buffer to hw buffer.

Reply:
uart_xmitchars copies the characters from the software tx buffer to the tx fifo. We are capturing the data when it is being added to the software buffer. So it should cover all the logs that are transmitted through the software buffer but the logs like lldbg will be left out as they don't use software buffer.